neuraxle/steps/util.py

```diff
@@ -217,16 +236,30 @@ def fit(self, data_inputs: List, expected_outputs: List = None) -> 'StepClonerFo
 
     def transform(self, data_inputs: List) -> List:
         # As many data inputs than we have cloned steps: each transforms the one it has.
-        assert len(data_inputs) <= len(self.steps), "Can't have more data_inputs than cloned steps to process them."
+        assert len(data_inputs) >= len(self.steps), "Can't have more data_inputs than cloned steps to process them."
 
         return [self.steps[i].transform(di) for i, di in enumerate(data_inputs)]
 
     def inverse_transform(self, data_output):
         # As many data outputs than we have cloned steps: each transforms the one it has.
-        assert len(data_output) <= len(self.steps), "Can't have more data_outputs than cloned steps to process them."
+        assert len(data_output) >= len(self.steps), "Can't have more data_outputs than cloned steps to process them."
 
         return [self.steps[i].inverse_transform(di) for i, di in enumerate(data_output)]
 
+    def set_hyperparams(self, hyperparams: HyperparameterSamples) -> BaseStep:
+        self.step = self.step.set_hyperparams(hyperparams.to_flat())
+        return self
```

Comment:
Would be a bit safer to add this too: 
`self.steps = [s.set_hyperparams(hyperparams.to_flat()) for s in self.steps]`

Reply:
Note that I'd prefer different commit (or perhaps even different PRs) for each issue. With git commit comments that fixes the issue number and links to it.

Reply:
not supposed to be in this pr

Reply:
It is said in PR #62 that it fixes this issue #29, however it is in the present PR #68 that I really see the diff for this fix of #29. I will add a comment for #69 in PR #62 then. You can close here, perhaps address the change in the other PR then.